### PR TITLE
Synonym Offenses

### PR DESCRIPTION
**Pull request by JoeCohen**
This PR fixes the last two Rubocop offenses in `synonym.rb`.

(Those offenses are normally hidden by `.rubocop_todo.yml`, but I noticed them while working on [Remove mass_insert gem #965](https://github.com/MushroomObserver/mushroom-observer/pull/965) and meant to correct them there.)

### CODE REVIEW

File: app/models/synonym.rb [hunk 1]
```diff
@@ -25,26 +25,32 @@
...
   def self.make_sure_all_referenced_synonyms_exist
-    msgs = []
     references = Name.select(:synonym_id).where.not(synonym: nil).distinct.
                  order(:synonym_id).pluck(:synonym_id)
     records = Synonym.all.order(id: :asc).pluck(:id)
     unused  = records - references
     missing = references - records
+
+    Synonym.where(id: unused).delete_all
+    Synonym.insert_all(missing.map { |m| Hash[id: m] })
```
Comment (JoeCohen) [inline app/models/synonym.rb:43]: Bad.
This will throw an error if nothing is missing.

**Comment by pellaea** [comment]
Looks good to me, thanks, Joe.

**Comment by nimmolo** [comment]
Looks good to me too.

**Comment by JoeCohen** [comment]
Thanks Nimmo!

**Pull request merged**